lib/rex/parser/arguments.rb

```diff
@@ -16,12 +16,16 @@ class Arguments
       # Initializes the format list with an array of formats like:
       #
       # Arguments.new(
-      #    '-b' => [ false, "some text" ]
+      #    '-b'                => [ false, "some text"                 ],
+      #    ['-b']              => [ false, "some text"                 ],
+      #    '--sample'          => [ false, "sample long arg"           ],
+      #    '--also-a-sample'   => [ false, "sample longer arg"         ],
+      #    ['-x', '--execute'] => [ true, "mixing long and short args" ]
       # )
       #
       def initialize(fmt)
         self.fmt = fmt
-        self.longest = fmt.keys.max_by(&:length)
+        self.longest = fmt.keys.map { |key| [key].flatten.join(', ') }.max_by(&:length)
```

Comment:
It may be easier to take `fmt` and normalize the option key to always be an array, that way you don't have to special case the key being an array or string elsewhere in your code:

```suggestion
        self.fmt = fmt.map { |key, metadata| [Array(key), metadata] }.to_h
        self.longest = self.fmt.keys.map { |key| key.flatten.join(', ') }.max_by(&:length)
```